pkg/server/secrets-encrypt.go

```diff
@@ -0,0 +1,508 @@
+package server
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/rancher/k3s/pkg/cluster"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/wrangler/pkg/generated/controllers/core"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+const (
+	EncryptionStart     string = "start"
+	EncryptionPrepare   string = "prepare"
+	EncryptionRotate    string = "rotate"
+	EncryptionReencrypt string = "reencrypt"
+)
+
+const aescbcKeySize = 32
+
+type EncryptionState struct {
+	Stage        string   `json:"stage"`
+	ActiveKey    string   `json:"activekey"`
+	Enable       int      `json:"enable,omitempty"` // -1: no config file, 0: disabled, 1: enabled
```

Comment:
nit: this is the specific use case for BoolPtr. nil = not configured, false = disabled, true = enabled.

Reply:
Swapped out for boolptr